duplex_http_call.go

```diff
@@ -306,6 +306,11 @@ func (d *duplexHTTPCall) makeRequest() {
 	// pipe. Write's check for io.ErrClosedPipe and will convert this to io.EOF.
 	response, err := d.httpClient.Do(d.request) //nolint:bodyclose
 	if err != nil {
+		if errors.Is(err, io.EOF) {
+			// We use io.EOF as a sentinel in many places and don't want this
+			// transport error to be confused for those other situations.
+			err = io.ErrUnexpectedEOF
+		}
```

Comment:
Could be moved as a `wrapIf...` for consistency?

Reply:
But we're not wrapping -- we are replacing.

Reply:
The error could be wrapped and this avoids downstream errors mutating it. All these errors get wrapped by `NewError(CodeUnavailable, err)`

Reply:
> All these errors get wrapped by `NewError(CodeUnavailable, err)`

I'm not really sure what you're saying. At this point, the `err` is directly from the `HTTPClient.Do` method -- no other Connect wrapping has occurred.

Reply:
I'm referring to the error logic that happens below. Each wrap function checks the error for a connect error, then if not a connect error attempts to match it. Setting it as a connect error upfront would avoid any other error wrapper from wrapping it as a different error.

Reply:
For example `wrapIfContextError` will check with `asError(err)` and then return early before attempting to wrap as a different error. https://github.com/connectrpc/connect-go/blob/99d6b9ceaf98d18c295f66af16acde0e639caca1/error.go#L302-L304